ecosystem/indexer-grpc/indexer-grpc-data-service/src/service.rs

```diff
@@ -512,28 +604,83 @@ fn get_request_metadata(req: &Request<GetTransactionsRequest>) -> tonic::Result<
         .collect();
     let request_metadata: RequestMetadata =
         serde_json::from_str(&serde_json::to_string(&request_metadata_map).unwrap()).unwrap();
-    // TODO: update the request name if these are internal requests.
+    let request_metadata = RequestMetadata {
+        request_connection_id: Some(Uuid::new_v4().to_string()),
+        ..request_metadata
+    };
     Ok(request_metadata)
 }
 
 async fn channel_send_multiple_with_timeout(
     resp_items: Vec<TransactionsResponse>,
     tx: tokio::sync::mpsc::Sender<Result<TransactionsResponse, Status>>,
+    current_batch_start_time: Instant,
 ) -> Result<(), SendTimeoutError<Result<TransactionsResponse, Status>>> {
+    // Evnet: num of chunks.
+    //      preparing data for chunking before sending to client.
+    let overall_size = resp_items
+        .iter()
+        .map(|resp_item| resp_item.encoded_len())
+        .sum::<usize>();
+    let start_version = resp_items.first().unwrap().transactions.first().unwrap().version;
+    let end_version = resp_items.last().unwrap().transactions.last().unwrap().version;
     for resp_item in resp_items {
-        let current_instant = std::time::Instant::now();
+        let response_size = resp_item.encoded_len();
+        let num_of_transactions = resp_item.transactions.len();
+        let start_version = resp_item.transactions.first().unwrap().version;
+        let end_version = resp_item.transactions.last().unwrap().version;
+        let start_version_txn_timestamp_in_sec = {
+            let timestamp = resp_item
+                .transactions
+                .first()
+                .unwrap()
+                .timestamp
+                .as_ref()
+                .unwrap();
+            timestamp.seconds as f64 + timestamp.nanos as f64 / 1_000_000_000.0
+        };
+        let end_version_txn_timestamp_in_sec = {
+            let timestamp = resp_item
+                .transactions
+                .last()
+                .unwrap()
+                .timestamp
+                .as_ref()
+                .unwrap();
+            timestamp.seconds as f64 + timestamp.nanos as f64 / 1_000_000_000.0
+        };
         tx.send_timeout(
             Result::<TransactionsResponse, Status>::Ok(resp_item),
             RESPONSE_CHANNEL_SEND_TIMEOUT,
         )
         .await?;
-        sample!(
-            SampleRate::Duration(Duration::from_secs(60)),
-            info!(
-                "[data service] response waiting time in seconds: {}",
-                current_instant.elapsed().as_secs_f64()
-            );
+        info!(
+            start_version = start_version,
+            end_version = end_version,
+            start_version_txn_timestamp = start_version_txn_timestamp_in_sec,
+            end_version_txn_timestamp = end_version_txn_timestamp_in_sec,
```

Comment:
We can also add txn_tps here since all the data is available.

Reply:
done.